packages/common/src/discussions/utils.ts

```diff
@@ -43,13 +49,101 @@ export function setDiscussableKeyword(
   return updatedTypeKeywords;
 }
 
+/**
+ * Determines if the given channel is considered to be a `public` channel, supporting both
+ * legacy permissions and V2 ACL model.
+ * @param channel An IChannel record
+ * @returns true if the channel is considered `public`
+ */
+export function isPublicChannel(channel: IChannel): boolean {
+  return channel.channelAcl
+    ? channel.channelAcl.some(
+        ({ category }) => category === AclCategory.AUTHENTICATED_USER
+      )
+    : channel.access === SharingAccess.PUBLIC;
+}
+
+/**
+ * Determines if the given channel is considered to be an `org` channel, supporting both
+ * legacy permissions and V2 ACL model.
+ * @param channel An IChannel record
+ * @returns true if the channel is considered `org`
+ */
+export function isOrgChannel(channel: IChannel): boolean {
+  return channel.channelAcl
+    ? channel.channelAcl.some(({ category }) => category === AclCategory.ORG)
+    : channel.access === SharingAccess.ORG;
+}
+
+/**
+ * Determines if the given channel is considered to be a `private` channel, supporting both
+ * legacy permissions and V2 ACL model.
+ * @param channel An IChannel record
+ * @returns true if the channel is considered `private`
+ */
+export function isPrivateChannel(channel: IChannel): boolean {
+  return !isPublicChannel(channel) && !isOrgChannel(channel);
```

Comment:
It may be better here to check that every channelAcl is only of category `group`

```js
channel.channelAcl.every(({ category }) => category === AclCategory.GROUP)
```